core/server/proxy/src/main/java/alluxio/proxy/s3/CompleteMultipartUploadHandler.java

```diff
@@ -164,11 +202,20 @@ public CompleteMultipartUploadResult call() throws S3Exception {
           }
         }
 
+        // Remove the temporary directory containing the uploaded parts and the
+        // corresponding Alluxio S3 API metadata file
         mFileSystem.delete(multipartTemporaryDir,
             DeletePOptions.newBuilder().setRecursive(true).build());
-        MultipartUploadCleaner.cancelAbort(mFileSystem, mBucket, mObject, mUploadId);
+        mFileSystem.delete(new AlluxioURI(
+            S3RestUtils.getMultipartMetaFilepathForUploadId(mUploadId)),
+            DeletePOptions.newBuilder().build());
+        if (mMultipartCleanerEnabled) {
```

Comment:
without understanding the entire PR, could you describe under what condition is the cancelAbort triggered? Because I thought getting here means we are completing multipart upload already?

Reply:
The `cancelAbort()` is referring to "abort" the existing background task which will try to `AbortMultipartUpload` long-running multipart uploads. So it's removing that queued task when its corresponding upload gets completed.

I agree the flow is a bit confusing because of the additions of the "Handler" & "Cleaner". They were merged in from the community so we don't have internal documentation (yet).